File: Festive discount.py
n = int(input())
t = sum((p := float(input())) * (1 - (0.5 if p >= 1000 else 0.25 if p > 500 else 0.2 if p > 400 else 0)) for _ in range(n))
t -= 100 if t > 2000 else 0
print(f"{t:.2f}")

Input
3
1000
1200
400
Output:
1500.00
Example 2:
Input:
5
450
500
2000
1600
300
Output:
2760.00

File: Festive discount.cpp
#include <iostream>
#include <iomanip>

int main() {
    int n;
    std::cin >> n;
    
    double t = 0;
    for (int i = 0; i < n; ++i) {
        double p;
        std::cin >> p;
        double d = (p >= 1000) ? 0.5 : (p > 500) ? 0.25 : (p > 400) ? 0.2 : 0;
        t += p * (1 - d);
    }
    
    if (t > 2000) {
        t -= 100;
    }
    
    std::cout << std::fixed << std::setprecision(2) << t << std::endl;

    return 0;
}

Input
3
1000
1200
400
Output:
1500.00
Example 2:
Input:
5
450
500
2000
1600
300
Output:
2760.00
